### DISCUSSION TITLE
use application password to prevent 2FA on (subsequent) executions

### DISCUSSION THREAD

**Discussion by OWL16** [icloud-photos-downloader/icloud_photos_downloader#753] [Ideas]
iCloud allows for generating an application-specific password. Intended to prevent 2FA for applications like mail clients, such an application password works fine with icloudpd as well.

**Comment by OWL16** [comment]
CORRECTION: seems to work, but I'm not sure it will stay working though...

**Reply by w-gitops** [reply]
I couldn't get it to work for me. did you use the same email as the primary icloud account?

**Comment by liteowl** [comment]
Is this still working for you? I got my sync running fine, using password and 2FA. I would like to switch to an app specific password. How do I get it to delete my original password from the Keyring?

**Reply by OWL16** [reply]
Nope. Application password required a 2FA after approx 4-6 weeks again. So no full unattended automation seems possible this way, unless you’re willing to reauth uour family member’s apple id’s every month.

**Reply by liteowl** [reply]
Ah, that's too bad. Doing 2FA every month is going to kill me, even if it's just for one account. Thanks for letting me know.

**Reply by boredazfcuk** [reply]
Application passwords only work on iDevices. It's not possible to use them to log into the iCloud website.

Also, renewal is only needed every 90 days.

I have 2-way comms working in my container, so I can force a sync from my phone. If it was possible to feed in the MFA code to icloudpd via text file, then I could do remote renewal too.

Just not really had the time, or motivation to look into it further really.

**Reply by liteowl** [reply]
Thanks for the info. I was asked to renew 2FA after 30 days. Will it go to 90 days after the 2nd time? 
(I cannot use Docker on my NAS so I'm not able to work in a container; that doesn't seem like it should affect the 2FA renewal timing though?)

**Reply by boredazfcuk** [reply]
30 days expiry is not normal. It should last 90 days every time.

Occasionally I get forced to renew in a smaller timeframe, but I am unsure as to the reason. I think it's basically whatever reason Apple wants.

Sometimes all four of my containers will stop syncing at the same time. I'm assuming a change at Apple's end has force-expired everyone's cookies.

Other times, I will get an email notification about my account being locked and I have to unlock it, then re-auth. I put this down to brute force attempts using an old leaked password.

Other times, when I'm testing the container and performing lots of syncs in a short timeframe, it will force a re-auth. That's probably just Apple policy.

I just sync twice a day now and it usually lasts the full 90.

**Reply by liteowl** [reply]
Good to know. Hoping this time will last 90 days! That would be a big improvement :-)
